test/src/test/java/jenkins/model/lazy/LazyBuildMixInTest.java

```diff
@@ -87,6 +90,40 @@ public class LazyBuildMixInTest {
         assertEquals(b3, b1a.getNextBuild());
     }
 
+    @Test public void numericLookups() throws Exception {
+        var p = r.createFreeStyleProject();
+        var b1 = r.buildAndAssertSuccess(p);
+        var b2 = r.buildAndAssertSuccess(p);
+        var b3 = r.buildAndAssertSuccess(p);
+        var b4 = r.buildAndAssertSuccess(p);
+        var b5 = r.buildAndAssertSuccess(p);
+        var b6 = r.buildAndAssertSuccess(p);
+        b1.delete();
+        b3.delete();
+        b6.delete();
+        // leaving 2, 4, 5
+        assertThat(p.getFirstBuild(), is(b2));
+        assertThat(p.getLastBuild(), is(b5));
+        assertThat(p.getNearestBuild(-1), is(b2));
+        assertThat(p.getNearestBuild(0), is(b2));
+        assertThat(p.getNearestBuild(1), is(b2));
+        assertThat(p.getNearestBuild(2), is(b2));
+        assertThat(p.getNearestBuild(3), is(b4));
+        assertThat(p.getNearestBuild(4), is(b4));
+        assertThat(p.getNearestBuild(5), is(b5));
+        assertThat(p.getNearestBuild(6), nullValue());
+        assertThat(p.getNearestBuild(7), nullValue());
+        assertThat(p.getNearestOldBuild(-1), nullValue());
+        assertThat(p.getNearestOldBuild(0), nullValue());
```

Comment:
The original implementation in fact called `getNearestOldBuild(0)`, which unconditionally returns `null`. You can deduce this from the original Javadoc but the default implementation (overridden in `LazyBuildMixIn`) based on `TailMap` was potentially confusing, since the map in question is sorted in reverse order. I noticed there was no direct test coverage of these methods and decided to add some to clarify the meaning.